src/main/java/nextstep/subway/domain/Section.java

```diff
@@ -0,0 +1,104 @@
+package nextstep.subway.domain;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "SECTION")
+public class Section extends BaseEntity {
+
+    private static int MIN_DISTANCE = 1;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "up_station_id")
+    private Station upStation;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "down_station_id")
+    private Station downStation;
+    private Integer distance;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "line_id")
+    private Line line;
+
+    protected Section() {}
+
+    private Section(Long id, Station upStation, Station downStation, Integer distance, Line line) {
+        validateStations(upStation, downStation);
+        validateDistance(distance);
+        this.id = id;
+        this.upStation = upStation;
+        this.downStation = downStation;
+        this.distance = distance;
+        this.line = line;
+    }
+
+    public static Section of(Station upStation, Station downStation, Integer distance) {
+        return new Section(null, upStation, downStation, distance, null);
+    }
+
+    public static Section of(Station upStation, Station downStation, Integer distance, Line line) {
+        return new Section(null, upStation, downStation, distance, line);
+    }
+
+    private void validateStations(Station... stations) {
+        if (stations == null || Arrays.stream(stations).anyMatch(Objects::isNull)) {
```

Comment:
```suggestion
        if (Objects.isNull(stations) || Arrays.stream(stations).anyMatch(Objects::isNull)) {
```
이 부분은 중요한 것은 아니지만 이렇게 써줄 수도 있을 것 같아요..!